Src/Support/Google.Apis.Auth/OAuth2/GoogleCredential.cs

```diff
@@ -290,6 +290,13 @@ Task<string> ITokenAccess.GetAccessTokenForRequestAsync(string authUri, Cancella
                 new AccessTokenWithHeaders(await credential.GetAccessTokenForRequestAsync(authUri, cancellationToken).ConfigureAwait(false));
         }
 
+        /// <inheritdoc/>
+        public Task<OIdCToken> GetOIdCTokenAsync(OIdCTokenOptions options, CancellationToken cancellationToken = default) =>
+            (UnderlyingCredential is IOIdCTokenProvider provider) ?
+            provider.GetOIdCTokenAsync(options, cancellationToken) :
+            throw new InvalidOperationException(
+                $"{nameof(UnderlyingCredential)} is not an OIDC token provider. Only {nameof(ServiceAccountCredential)}, {nameof(ComputeCredential)}, {nameof(UserCredential)} are supported OIDC token providers.");
```

Comment:
Just as a reminder, this will need changing once UserCredential isn't an OIDC provider.